components/camel-consul/src/test/java/org/apache/camel/component/consul/ConsulRegistryTest.java

```diff
@@ -49,19 +55,26 @@ public String hello(String name) {
         }
     }
 
+   
     @BeforeClass
     public static void setUp() {
-        container = ConsulTestSupport.consulContainer();
-        container.start();
+        if (DockerMachineClient.instance().isInstalled())  {
```

Comment:
@ramu11 , can we move this check to the camel-testcontainers level to prevent using this logic in each component?

Reply:
+1

Reply:
This class is not extending 'camel-testcontainers' test class

Reply:
This class do not extends camel-testcontainers test class. And the class has both @BeforeClass and @AfterClass methods where the solution do not work. I already tested the scenario

Reply:
@ramu11 , in this case, may be create a super abstract class for this in camel-testcontainers?